numba/np/arrayobj.py

```diff
@@ -1394,6 +1394,52 @@ def impl(array, shape):
     return impl
 
 
+if numpy_version >= (1, 20):
+    @overload(np.broadcast_shapes)
+    def numpy_broadcast_shapes(*args):
+        # Based on https://github.com/numpy/numpy/blob/f702b26fff3271ba6a6ba29a021fc19051d1f007/numpy/core/src/multiarray/iterators.c#L1129-L1212  # noqa
```

Comment:
> This link is for `PyArray_Broadcast` ? I'm guessing this is because of no `np.broadcast` support in Numba as yet?

No, NumPy calls this code internally. See https://github.com/numpy/numpy/blob/4adc87dff15a247e417d50f10cc4def8e1c17a03/numpy/lib/stride_tricks.py#L415-L429



> Also, having looked at the C code, it's very loosely based on the branching/computation of shape?

Yes, that's right.

Reply:
Yes to both questions :smile: